File: src/Sample.Assembly.One/CarViewModel.cs
using System;
using TypeScript.Modeller;

namespace Sample.Assembly.One
{
    [TypeScriptViewModel]
    public class CarViewModel
    {
        public string StringWithLongCamelCase { get; set; }

        public Single SingleDataType { get; set; }

        public long? Year { get; set; }

        public PersonViewModel Owner { get; set; }

        public RegistrationViewModel Registration { get; set; }
    }
}

File: src/TypeScript.Modeller.Tests.Integration/SampleOneExpected/CarViewModel.ts
import { PersonViewModel } from './PersonViewModel';
import { RegistrationViewModel } from './RegistrationViewModel';

export class CarViewModel {
    public stringWithLongCamelCase: string = '';
    public singleDataType: number = 0;
    public year?: number = undefined;
    public owner: PersonViewModel | undefined = undefined;
    public registration: RegistrationViewModel | undefined = undefined;
}
